components/image/slider/src/index.js

```diff
@@ -13,99 +12,97 @@ export const IMAGE_SLIDER_COUNTER_POSITIONS = {
-            <ReactSlidy {...this._sliderOptions}>{slides}</ReactSlidy>
-          ) : (
-            slides
-          )}
-          {enableCounter && this._buildCounter(slides.length)}
-        </div>
-      )
+    const key = imageKey ? imageKey + index : index
+    const img = (
+      <img
+        alt={alt}
+        className="sui-ImageSlider-image"
+        key={key}
+        src={src}
+        title={title}
+      />
     )
-  }
+    return link
+      ? linkFactory({
+          key,
+          target,
+          className: '',
+          children: img,
+          href: link
+        })
+      : img
+  })
+}
+
+export default function ImageSlider({
+  images = [],
+  handleClick = NO_OP,
+  sliderOptions = {},
+  linkFactory = defaultLinkFactory,
+  enableCounter = false,
+  counterPosition = IMAGE_SLIDER_COUNTER_POSITIONS.BOTTOM_RIGHT,
+  counterIcon = IconCamera,
+  counterPatternFactory = defaultCounterPatternFactory
+}) {
+  const [currentSlide, setCurrentSlide] = useState(
+    sliderOptions.initialSlide || 0
+  )
+  const slides = getSlides(images, linkFactory)
 
-  _buildCounter(totalImages) {
+  const buildCounter = totalImages => {
     const classNames = cx(
       'sui-ImageSlider-counter',
-      `sui-ImageSlider-counter--${this.props.counterPosition}`
+      `sui-ImageSlider-counter--${counterPosition}`
     )
-    const Icon = this.props.counterIcon
+    const Icon = counterIcon
     return (
       <div className={classNames}>
         <Icon svgClass="sui-ImageSlider-counterIcon" />
         <span className="sui-ImageSlider-counterText">
-          {this.props.counterPatternFactory({
-            current: this.state.currentSlide + 1,
+          {counterPatternFactory({
+            current: currentSlide + 1,
             total: totalImages
           })}
         </span>
       </div>
     )
   }
 
-  /**
-   * @param {Array} images List given by props.images.
-   * @return {Array} List of img elements.
-   */
-  _getSlides(images, linkFactory) {
-    if (images && images.length) {
-      return images.map((image, index) => {
-        const {
-          key: imageKey,
-          alt,
-          link,
-          src,
-          target = TARGET_BLANK,
-          title
-        } = image
-
-        const key = imageKey ? imageKey + index : index
-        const img = (
-          <img
-            alt={alt}
-            className="sui-ImageSlider-image"
-            key={key}
-            src={src}
-            title={title}
-          />
-        )
-        return link
-          ? linkFactory({
-              key,
-              target,
-              className: '',
-              children: img,
-              href: link
-            })
-          : img
-      })
-    } else {
-      return []
-    }
+  const handleAfterSlide = ({currentSlide}) => {
+    setCurrentSlide(currentSlide)
+    sliderOptions.doAfterSlide && sliderOptions.doAfterSlide(currentSlide)
   }
+
+  return (
+    slides.length > 0 && (
```

Comment:
It would be useful to define this logic in a constant?

Reply:
Well, I wanted to avoid touching as much as possible the actual code. Anyway, it seems to be readable enough! Not perfect tough.